### PR TITLE
Make SignTool accept empty PKT on FileSignInfo

### PR DESCRIPTION
**Pull request by JohnTortugo**
Closes: #496 

**Changes:**

- SignToolTask.cs: 
  - Accept empty/null metadata on FileSignInfo->PublicKeyToken; 
  - Don't .ToLower() on null
- Configuration.cs:
  - Don't fail when public key token is empty/null
- SignToolTests.cs:
  - Added new test case for scenario where file to be signed doesn't have PKT
- Resources/EmptyPKT.dll
  - Test case input file without a PKT

### CODE REVIEW

File: src/Microsoft.DotNet.SignTool/src/Configuration.cs [hunk 1]
```diff
@@ -141,6 +141,17 @@ private SignInfo ExtractSignInfo(string fileFullPath)
                         // Otherwise, just use the overriding info if present
                     }
 
+                    if (string.IsNullOrEmpty(publicKeyToken))
```
Comment (tmat) [inline src/Microsoft.DotNet.SignTool/src/Configuration.cs]: >IsNullOrEmpty [](start = 31, length = 13)

publicKeyToken can't be null here.

File: src/Microsoft.DotNet.SignTool/src/Configuration.cs [hunk 2]
```diff
@@ -121,16 +121,16 @@ private SignInfo ExtractSignInfo(string fileFullPath)
                 else
                 {
                     var fileAsm = System.Reflection.AssemblyName.GetAssemblyName(fileFullPath);
-                    var publicKeyToken = string.Join("", fileAsm.GetPublicKeyToken().Select(b => b.ToString("x2")));
+                    var pktBytes = fileAsm.GetPublicKeyToken();
```
Comment (tmat) [inline src/Microsoft.DotNet.SignTool/src/Configuration.cs]: >pktBytes [](start = 24, length = 8)

This can be null.

**Comment by tmat** [comment]
Test?

File: src/Microsoft.DotNet.SignTool/src/Configuration.cs (hunk 2 above)
Comment (JohnTortugo) [reply src/Microsoft.DotNet.SignTool/src/Configuration.cs]: Local tests only resulted in an empty array of bytes. Still, I added the null check to be on the safe side.

File: src/Microsoft.DotNet.SignTool/src/Configuration.cs (hunk 1 above)
Comment (JohnTortugo) [reply src/Microsoft.DotNet.SignTool/src/Configuration.cs]: It was matching == "". Changed to be more explicit.

File: tests/Microsoft.DotNet.SignTool.Tests/SignToolTests.cs [hunk 3]
```diff
@@ -174,7 +174,34 @@ public void OnlyContainerAndOverriding()
...
+                $@"Resources/EmptyPKT.dll",
+            };
+
+            // Default signing information
+            var signingInformation = new Dictionary<string, SignInfo>() {
+                { "581d91ccdfc4ea9c", new SignInfo("ArcadeCertTest", "ArcadeStrongTest") }
+            };
+
+            // Overriding information
+            var signingOverridingINformation = new Dictionary<ExplicitCertificateKey, string>() {
```
Comment (maririos) [inline tests/Microsoft.DotNet.SignTool.Tests/SignToolTests.cs]: NIT: signingOverridingINformation  => signingOverridingInformation

**Comment by natemcmaster** [comment]
Thanks for getting to this so quickly @JohnTortugo :)

**Pull request merged**